ocsp/responder.go

```diff
@@ -243,12 +261,21 @@ func (rs Responder) ServeHTTP(response http.ResponseWriter, request *http.Reques
 	}
 
 	// Look up OCSP response from source
-	ocspResponse, found := rs.Source.Response(ocspRequest)
-	if !found {
-		log.Infof("No response found for request: serial %x, request body %s",
-			ocspRequest.SerialNumber, b64Body)
-		response.Write(unauthorizedErrorResponse)
-		return
+	ocspResponse, extraHeaders, err := rs.Source.Response(ocspRequest)
+	if err != nil {
+		if err == ErrNotFound {
+			if extraHeaders != nil {
+				writeExtraHeaders(response, extraHeaders)
```

Comment:
Go idiom is to avoid returning meaningful values when an err return is non-nil, so we should probably ignore the extraHeaders on error returns.